File: scratch/picuki.js
const fs = require('fs');
const https = require('https');

function downloadImage(url, filepath) {
    return new Promise((resolve, reject) => {
        https.get(url, (res) => {
            if (res.statusCode === 200) {
                res.pipe(fs.createWriteStream(filepath))
                   .on('error', reject)
                   .once('close', () => resolve(filepath));
            } else {
                res.resume();
                reject(new Error(`Request Failed With a Status Code: ${res.statusCode}`));
            }
        });
    });
}

async function scrape() {
    try {
        const res = await fetch("https://www.picuki.com/profile/spicesalon.mn", {
            headers: {
                "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64)"
            }
        });
        const html = await res.text();
        
        const profileMatch = html.match(/<img[^>]*class="profile-avatar"[^>]*src="([^"]+)"/);
        if (profileMatch) {
            console.log("Found logo:", profileMatch[1]);
            await downloadImage(profileMatch[1], "C:\\Users\\syclo\\retrofit-web\\spice-salon-mockup\\public\\logo.jpg");
        }
        
        const postMatches = [...html.matchAll(/<img[^>]*class="post-image"[^>]*src="([^"]+)"/g)];
        for (let i = 0; i < Math.min(2, postMatches.length); i++) {
            const url = postMatches[i][1];
            console.log("Found post:", url);
            const name = i === 0 ? "hero.jpg" : "service.jpg";
            await downloadImage(url, `C:\\Users\\syclo\\retrofit-web\\spice-salon-mockup\\public\\${name}`);
        }
        console.log("Done");
    } catch (e) {
        console.error("Error:", e);
    }
}

scrape();

File: scratch/picuki.py
import urllib.request
import re
import os

url = "https://www.picuki.com/profile/spicesalon.mn"
req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36'})
try:
    html = urllib.request.urlopen(req).read().decode('utf-8')
    # Find profile picture
    profile_match = re.search(r'<img[^>]*class="profile-avatar"[^>]*src="([^"]+)"', html)
    if profile_match:
        profile_url = profile_match.group(1)
        print("Found logo:", profile_url)
        urllib.request.urlretrieve(profile_url, r"C:\Users\syclo\retrofit-web\spice-salon-mockup\public\logo.jpg")
    
    # Find post images
    post_matches = re.findall(r'<img[^>]*class="post-image"[^>]*src="([^"]+)"', html)
    for i, post_url in enumerate(post_matches[:2]):
        print(f"Found post {i}:", post_url)
        name = "hero.jpg" if i == 0 else "service.jpg"
        urllib.request.urlretrieve(post_url, rf"C:\Users\syclo\retrofit-web\spice-salon-mockup\public\{name}")
        
    print("Done")
except Exception as e:
    print("Error:", e)
